### DISCUSSION TITLE
Changing to parcel from webpack with bare specifiers

### DISCUSSION THREAD

**Discussion by richardkmichael** [parcel-bundler/parcel#8697] [Q&A] [answered]
I am migrating a webpack 4 project to Parcel.

The project uses webpack's [`resolve.modules`](https://v4.webpack.js.org/configuration/resolve/#resolvemodules) setting to prepend to bare specifiers when searching for modules.

`webpack.config.js`:

```javascript
module.exports = {
  resolve: {
    modules: [
      "src"
    ]
  }
};
```

So that webpack will prepend `src/` to import paths when resolving a bare specified name, e.g.:

In `src/path/to/this-file.js`
```javascript
import { myFunc } from 'myModule'
```

.. webpack will find `src/myModule/index.js`.

I don't believe this causes webpack to _search recursively_, it only allows the module name to be written without the `src/` directory component.

I've read the Parcel dependency and module resolution docs, and I don't see this covered.  I also looked through GitLab Discussions and Issues.  Sorry if I've missed something.

Should I write a small Parcel Resolver plugin?

Eventually, I would prefer the module names to be qualified, either absolute or relative, so that resolution is faster (less searching) and the build time is faster.  I think the most efficient way to do this will be to write a webpack (or parcel plugin) to modify and rewrite the original source code during a on-off "fake" build (i.e., after resolving, rewrite the original source with the found location and exit).

Thoughts on rewriting imports would be welcome too.

Thanks!

**Comment by mischnic** [comment, marked as answer]
How many of these pseudo-modules do you have? Creating aliases might work: https://parceljs.org/features/dependency-resolution/#aliases

To recreate that Webpack feature, then yes, you need a resolver plugin.

**Reply by richardkmichael** [reply]
There are about 2000 files using `import ...` in that manner.  I'm unsure how many targets there are, but many.

I noticed `aliases`, but yes, creating so many of them didn't seem like a feasible approach.

I'll have a look at writing a plugin, this situation seems fairly straight forward.

Thanks!
